Sources/SmithyReadWrite/DocumentReader.swift

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+
+public enum DocumentReader {
```

Comment:
This type was moved in from `ClientRuntime`, and was made generic to any `SmithyReader`.